src/instructlab/model/evaluate.py

```diff
@@ -213,29 +220,32 @@ def qa_pairs_to_qna_to_avg_scores(qa_pairs: list[dict]) -> dict:
     return qna_to_avg_scores
 
 
-def launch_server(
-    ctx,
-    model,
-    model_name,
-    max_workers,
-) -> tuple:
+def launch_server(ctx, model, model_name, max_workers, backend, max_ctx_size) -> tuple:
```

Comment:
@cdoern Sebastian has most of the infrastructure in place to autodetect what backend you're going to use depending on the path passed in via `--model`. Take a look at `src/instructlab/model/backends/backends.py` at the gguf autodetection code he already wrote. Khaled is working on something similar for .safetensors.

There may need to be a bit more work done but what if we remove the `--backend` flag entirely from the evaluate commands.

IMO the only command specifying `--backend` makes sense is for `ilab model serve` itself.

Reply:
Why would a user want to set a diff backend just for eval if it was already autodetected before? I follow your approach, and I'm good with it as long as that is a use-case that would be hit

Reply:
the way I was thinking of it @alinaryan is if a person wants their backend to be llama say for easy single GPU operation but then for eval they want to use vllm so that they can use multi gpu support they can do that.

More generally though, the approach we are taking for all other cmds is a flag that serves as an override for anything the user might want to change on a single run. IMO I am not a huge fan of autodetecting behind the scenes and I know users will most likely want to set this themselves in the future whether there be a bug with one of the backends or special features of one of the backends. 

Autodetection is great for a base case but we can't assume auto-detect will be a catch all for customer or community use cases I think

Reply:
That makes sense to me in the multi-gpu context! SGTM. I also agree that we should keep with the overrides especially if you have introduced that in the other cmds already. Let's be as consistent as possible :rocket:

Reply:
serve cfg is unchanged, will follow up with some work to patch this flag to the serve cfg